### ISSUE TITLE
Update Deliverable .spec.source description to match use case

### ISSUE THREAD

**Issue by waciumawanjohi** [vmware-tanzu/cartographer#647]
### Description of problem

The deliverable expects to receive k8s configuration through a gitops workflow. Hence it has a .spec.source field. But the [current description](https://github.com/vmware-tanzu/cartographer/blob/9f693fbea1110b1fa709d5a7a9f7e05293e7c204/pkg/apis/v1alpha1/deliverable.go#L90-L92) states that this should be source code.

### Proposed solution

Update the description

```gherkin
Given I am a new Carto user
When I read the deliverable CRD doc
Then I understand the use case for this field
```

TODO:
* [ ] Generate the docs

**Comment by tcdowney** [comment]
Thanks for creating this! Not in scope here, but a suggestion for some future docs would be an end-to-end diagram showing a build cluster using `SupplyChain`s and `Workload`s and ultimately pushing to the git repo. Then some run clusters using the `Delivery`/`Deliverable` resources to pull and run this. I think the [examples](https://github.com/vmware-tanzu/cartographer/tree/main/examples) are a step in the right direction, but that something visual would help me picture it all better. 👍

**Comment by waciumawanjohi** [comment]
Is this [the diagram](https://cartographer.sh/docs/v0.2.0/architecture/#complete-supply-chain-and-delivery-with-gitops) you were hoping for?
If so, sounds like we should do a better job of surfacing it (maybe just link it in the examples).

**Comment by tcdowney** [comment]
Yeah actually, that's pretty much it. 😁  I didn't think to click on "Architecture" on the side. I think maybe some more direct "Concepts and Usage" linking or tutorials or something would help. Linking to it from the examples would be helpful as well. 👍
